### PR TITLE
Add opnsense config role

### PR DESCRIPTION
**Pull request by DonGiovanni83**
This PR adds a general configuration role for OPNsense instances. The goal of this role is to handle all module invocation, based on the given host variables.
The usage of the role is documented in the role README.md

### CODE REVIEW

File: changelogs/fragments/155-add-opnsense-confi-role.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+---
```
Comment (KiLLuuuhh) [inline changelogs/fragments/155-add-opnsense-confi-role.yml:1]: I suggest rename this file to `changelogs/fragments/155-add-opnsense-config-role.yml`

File: roles/opnsense_configure/README.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
+opnsense_configure - OPNsense configuration role
+=========
+
+This role provides a generic approach to configure OPNsense instances by populating host variables
+according to this roles defaults specification.
+
+Role Variables
+--------------
+
+The variables must be structured in a way that each puzzle.opnsense module has its own block. Each module related block
```
Comment (ombre8) [inline roles/opnsense_configure/README.md:10]: Does Block make sense in this context? I, for one, thought about an Ansible Block, which was kind a confusing... maybe `dictionary` or `map` are better words than `block` :thinking:

File: roles/opnsense_configure/meta/main.yml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+galaxy_info:
+  author: Fabio Bertagna <bertagna@puzzle.ch>
+  company: Puzzle ITC
+  license: GPL-3.0-only
+  min_ansible_version: 2.1
+  galaxy_tags:
+    - opnsense
+
+dependencies: [ ]
```
Comment (ombre8) [inline roles/opnsense_configure/meta/main.yml:9]: wound't the module be dependency of the role

**Review by ombre8 — COMMENTED** [review]
Thanks

File: roles/opnsense_configure/meta/main.yml (hunk 3 above)
Comment (DonGiovanni83) [reply roles/opnsense_configure/meta/main.yml:9]: My understanding of the `dependency` block in this file is that you only need to list dependency to other roles. https://docs.ansible.com/ansible/latest/galaxy/user_guide.html#dependencies

File: roles/opnsense_configure/README.md (hunk 2 above)
Comment (DonGiovanni83) [reply roles/opnsense_configure/README.md]: You are right, this can be confusing. I opted for `variable section` to replace `block`.

**Review by ombre8 — APPROVED** [review]
LGTM

**Pull request merged**